### PR TITLE
Implement USB HID Device Support for ESP32-S2

### PR DESCRIPTION
**Pull request by me-no-dev**

### CODE REVIEW

**Comment by me-no-dev** [comment]
@pedrominatel @SuGlider @igrr @lbernstone @atanisoft PTAL

**Comment by SuGlider** [comment]
Nice! I'll test it.

**Comment by SuGlider** [comment]
Tested it with Windows 10.
LGTM

File: libraries/USB/src/USBHIDVendor.cpp [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+static uint8_t feature[64];
+static xQueueHandle rx_queue = NULL;
+static bool prepend_size = false;
+
+USBHIDVendor::USBHIDVendor(uint8_t report_size, bool prepend): hid(){
+	static bool initialized = false;
+	if(!initialized){
+		initialized = true;
+		hid.addDevice(this, TUD_HID_REPORT_DESC_GENERIC_INOUT_FEATURE_LEN);
+        memset(feature, 0, 64);
```
Comment (atanisoft) [inline libraries/USB/src/USBHIDVendor.cpp:69]: it looks like a mix of tab/spaces in this method.

File: libraries/USB/src/USBHID.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include <stdint.h>
+#include <stdbool.h>
+#include "esp32-hal.h"
+#include "USB.h"
+#if CONFIG_TINYUSB_HID_ENABLED
```
Comment (atanisoft) [inline libraries/USB/src/USBHID.h:20]: Any reason why both `CFG_TUD_HID` is checked in the cpp files but `CONFIG_TINYUSB_HID_ENABLED` is checked in headers? Shouldn't they be the same value?

Comment (me-no-dev) [reply libraries/USB/src/USBHID.h]: they generally should. done in https://github.com/espressif/arduino-esp32/pull/5538/commits/196ce947789c1d8f3a86047110049ab0edbef5ec

File: libraries/USB/src/USBHIDVendor.cpp (hunk 1 above)
Comment (me-no-dev) [reply libraries/USB/src/USBHIDVendor.cpp:70]: also fixed :)

**Pull request merged**